### PR TITLE
multiple sensitive features - postprocessing

### PR DESCRIPTION
**Pull request by romanlutz**
This is the trimmed down version of #280  for just postprocessing. After this is merged I'll update #280 which will then only be about reductions.

### CODE REVIEW

File: fairlearn/_input_validation.py [hunk 1]
```diff
@@ -3,18 +3,96 @@
...
+        distinct values in the `sensitive_features` data from `kwargs`; default False
+    :type enforce_binary_sensitive_feature: bool
+    :param enforce_binary_labels: if True raise exception if there are more than two distinct
+        values in the `y` data; default False
+    :type enforce_binary_labels: bool
+    """
+    if y is not None:
+        # calling check_X_y with a 2-dimensional y causes a warning, so ensure it is 1-dimensional
+        if isinstance(y, np.ndarray) and len(y.shape) == 2 and y.shape[1] == 1:
+            y = y.reshape(-1)
```
Comment (riedgar-ms) [inline fairlearn/_input_validation.py]: How about np.squeeze?

Comment (romanlutz) [reply fairlearn/_input_validation.py]: Sure we can do that!

File: test/unit/postprocessing/conftest.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+
+@pytest.fixture(params=candidate_A_transforms)
+def data_sf(data, request):
+    sensitive_feature_transform = request.param
+    # Skip combinations where the multi-column sensitive features would have to be compressed
+    # into a one-dimensional data structure.
+    if (data.sensitive_features == sensitive_features_ex3).all() and \
+            sensitive_feature_transform in [ensure_list_1d, ensure_series]:
+        pytest.skip(MULTIPLE_SENSITIVE_FEATURE_COMPRESSION_SKIP_REASON)
```
Comment (riedgar-ms) [inline test/unit/postprocessing/conftest.py]: Is there no way we can prevent the invalid combinations being generated? Seeing skipped tests makes me think we're either got unfixed bugs (and hence are suppressing the test until the fix is in) or unimplemented features (similarly).

File: fairlearn/_input_validation.py (hunk 1 above)
Comment (riedgar-ms) [reply fairlearn/_input_validation.py]: Just be aware that `squeeze` on a single element array results in a scalar.

File: test/unit/postprocessing/conftest.py (hunk 2 above)
Comment (romanlutz) [reply test/unit/postprocessing/conftest.py]: It's not super straightforward but I'll see what I can do

Comment (romanlutz) [reply test/unit/postprocessing/conftest.py]: Is this acceptable to you?
```
1488 passed, 192 deselected in 285.46 seconds

**Pull request merged**